### PR TITLE
API-11984 DatamartImmunization vaccineCode array

### PR DESCRIPTION
**Pull request by jblefkove-va**

### CODE REVIEW

**Comment by jblefkove-va** [comment]
FYI @nmoisan @jtaylor-va @haidong

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/Dstu2ImmunizationTransformer.java [hunk 1]
```diff
@@ -79,17 +81,21 @@ public Immunization toFhir() {
...
     return CodeableConcept.builder()
-        .text(vaccineCode.text())
         .coding(
             Coding.builder()
                 .system("http://hl7.org/fhir/sid/cvx")
-                .code(vaccineCode.code())
+                .code(coding.code().orElse(null))
                 .build()
                 .asList())
+        .text(coding.display().orElse(null))
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/Dstu2ImmunizationTransformer.java:98]: Should this go in the coding display as well?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunization.java [hunk 2]
```diff
@@ -213,8 +216,30 @@ private void setEtlDate(String unused) {
...
+    /** Lazy getter. */
+    public List<DatamartCoding> coding() {
+      if (coding == null) {
+        coding = new ArrayList<>();
+      }
+      return coding;
+    }
+
+    @SuppressWarnings("unused")
+    private void setCode(String s) {
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunization.java:230]: Does this get used?

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/DatamartImmunization.java:230]: backwards compatibility

**Review by taylorrassmannlibertyits — DISMISSED** [review]

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java [hunk 3]
```diff
@@ -116,17 +116,22 @@ Immunization toFhir() {
   }
 
   CodeableConcept vaccineCode(DatamartImmunization.VaccineCode vaccineCode) {
-    if (vaccineCode == null || allBlank(vaccineCode.text(), vaccineCode.code())) {
+    if (vaccineCode == null || isBlank(vaccineCode.coding())) {
+      return null;
+    }
+    var coding = vaccineCode.coding().get(0);
```
Comment (jblefkove-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/immunization/R4ImmunizationTransformer.java:122]: the full list of codings will be transformed in a future PR

**Review by jhulbert-va — APPROVED** [review]

**Pull request merged**